File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnVideoSeason.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* \class GnVideoSeason
*/
public class GnVideoSeason : GnDataObject {
  private HandleRef swigCPtr;

  internal GnVideoSeason(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnVideoSeason obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnVideoSeason() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnVideoSeason(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_GnType();
    return ret;
  }

  public static GnVideoSeason From(GnDataObject obj) {
    GnVideoSeason ret = new GnVideoSeason(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public GnVideoSeason(string id, string idTag) : this(gnsdk_csharp_marshalPINVOKE.new_GnVideoSeason(id, idTag), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Genre. This is a list/locale dependent, multi-level field.
*
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful
*  allocation of a <code>GnLocale</code> object
*
*  To render locale-dependent information for list-based values, the application must allocate a GnLocale object.
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for
*  a list-based value.
* @ingroup GDO_ValueKeys_Misc
*/
  public string Genre(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Genre(swigCPtr, (int)level);
    return ret;
  }

/**
*  Origin, e.g., "New York City." List/locale dependent, multi-level field.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful
*  allocation of a <code>GnLocale</code> object
*
*  To render locale-dependent information for list-based values, the application must allocate a GnLocale object.
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for
*  a list-based value.
* @ingroup GDO_ValueKeys_Misc
*/
  public string Origin(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Origin(swigCPtr, (int)level);
    return ret;
  }

/**
*  Video season's Gracenote identifier.
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string GnId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_GnId_get(swigCPtr) );
	} 

  }

/**
*   Video season's Gracenote unique identifier.
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string GnUId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_GnUId_get(swigCPtr) );
	} 

  }

/**
*  Video season's TUI (title-unique identifier).
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string Tui {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Tui_get(swigCPtr) );
	} 

  }

/**
*  Video season's Tui Tag.
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string TuiTag {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_TuiTag_get(swigCPtr) );
	} 

  }

/**
*  Video season's product ID (aka Tag ID).
*  <p><b>Remarks:</b></p>
*  Available for most types, this value can be stored or transmitted. Can
*  be used as a static identifier for the current content as it will not change over time.
* @ingroup GDO_ValueKeys_GracenoteIDs
*/
  public string ProductId {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_ProductId_get(swigCPtr) );
	} 

  }

/**
*  Video production type, e.g., Animation. This is a list/locale dependent value.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful
*  allocation of a GnLocale object
*
*  To render locale-dependent information for list-based values, the application must allocate a GnLocale object.
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for
*  a list-based value.
* @ingroup GDO_ValueKeys_Video
*/
  public string VideoProductionType {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_VideoProductionType_get(swigCPtr) );
	} 

  }

/**
*  Video production type identifier.
* @ingroup GDO_ValueKeys_Video
*/
  public uint VideoProductionTypeId {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_VideoProductionTypeId_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Video season's original release date.
* @ingroup GDO_ValueKeys_Video
*/
  public string DateOriginalRelease {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_DateOriginalRelease_get(swigCPtr) );
	} 

  }

/**
*  Duration in seconds such as "3600" for a 60-minute program.
* @ingroup GDO_ValueKeys_Video
*/
  public uint Duration {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Duration_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Duration units value (seconds, "SEC").
* @ingroup GDO_ValueKeys_Video
*/
  public string DurationUnits {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_DurationUnits_get(swigCPtr) );
	} 

  }

/**
*  Franchise number.
* @ingroup GDO_ValueKeys_Video
*/
  public uint FranchiseNum {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_FranchiseNum_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Franchise count.
* @ingroup GDO_ValueKeys_Video
*/
  public uint FranchiseCount {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_FranchiseCount_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Plot sypnosis, e.g., for Friends episode:."Monica's popularity at a karaoke club might have more to do with her revealing dress than her voice"
* @ingroup GDO_ValueKeys_Video
*/
  public string PlotSynopsis {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_PlotSynopsis_get(swigCPtr) );
	} 

  }

/**
*  Plot sypnosis language, e.g., English
* @ingroup GDO_ValueKeys_Video
*/
  public string PlotSynopsisLanguage {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_PlotSynopsisLanguage_get(swigCPtr) );
	} 

  }

/**
*  Plot tagline, e.g., "An adventure as big as life itself."
* @ingroup GDO_ValueKeys_Video
*/
  public string PlotTagline {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_PlotTagline_get(swigCPtr) );
	} 

  }

/**
*  Serial type, e.g., Series or Episode
* @ingroup GDO_ValueKeys_Video
*/
  public string SerialType {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SerialType_get(swigCPtr) );
	} 

  }

/**
*  Work type, e.g., Musical
* @ingroup GDO_ValueKeys_Video
*/
  public string WorkType {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_WorkType_get(swigCPtr) );
	} 

  }

/**
*  Target audience, e.g., "Kids and Family"
* @ingroup GDO_ValueKeys_Video
*/
  public string Audience {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Audience_get(swigCPtr) );
	} 

  }

/**
*  Video mood, e.g., Playful.
* @ingroup GDO_ValueKeys_Video
*/
  public string VideoMood {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_VideoMood_get(swigCPtr) );
	} 

  }

/**
*  Overall story type, e.g., "Love Story".
* @ingroup GDO_ValueKeys_Video
*/
  public string StoryType {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_StoryType_get(swigCPtr) );
	} 

  }

/**
*  Reputation, e.g., "Chick flick".
* @ingroup GDO_ValueKeys_Video
*/
  public string Reputation {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Reputation_get(swigCPtr) );
	} 

  }

/**
*  Scenario, e.g., Action
* @ingroup GDO_ValueKeys_Video
*/
  public string Scenario {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Scenario_get(swigCPtr) );
	} 

  }

/**
*  Setting environment, e.g., "High School"
* @ingroup GDO_ValueKeys_Video
*/
  public string SettingEnvironment {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SettingEnvironment_get(swigCPtr) );
	} 

  }

/**
*  Setting time period, e.g., "Jazz Age 1919-1929".
* @ingroup GDO_ValueKeys_Video
*/
  public string SettingTimePeriod {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SettingTimePeriod_get(swigCPtr) );
	} 

  }

/**
*  Topic value, e.g., "Teen Angst"
* @ingroup GDO_ValueKeys_Video
*/
  public string Topic {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Topic_get(swigCPtr) );
	} 

  }

/**
*  Season number.
* @ingroup GDO_ValueKeys_Video
*/
  public uint SeasonNumber {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SeasonNumber_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Season count - total number of seasons.
* @ingroup GDO_ValueKeys_Video
*/
  public uint SeasonCount {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_SeasonCount_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Source, e.g., "Fairy Tales and Nursery Rhymes"
* @ingroup GDO_ValueKeys_Video
*/
  public string Source {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Source_get(swigCPtr) );
	} 

  }

/**
*  Style, e.g., "Film Noir"
* @ingroup GDO_ValueKeys_Video
*/
  public string Style {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Style_get(swigCPtr) );
	} 

  }

/**
*  Flag indicating if response result contains full (true) or partial metadata.
* <p><b>Note:</b></p>
*   What constitutes a full result varies among response results, and also
*  depends on data availability.
* @ingroup GDO_ValueKeys_Misc
*/
  public bool IsFullResult {
    get {
      bool ret = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_IsFullResult_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Rating object.
* @ingroup GDO_ValueKeys_Misc
*/
  public GnRating Rating {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Rating_get(swigCPtr);
      GnRating ret = (cPtr == IntPtr.Zero) ? null : new GnRating(cPtr, true);
      return ret;
    } 
  }

/**
*   Official title object
* @ingroup GDO_ChildKeys_Title
*/
  public GnTitle OfficialTitle {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_OfficialTitle_get(swigCPtr);
      GnTitle ret = (cPtr == IntPtr.Zero) ? null : new GnTitle(cPtr, true);
      return ret;
    } 
  }

/**
*  Franchise title object.
* @ingroup GDO_ChildKeys_Video
*/
  public GnTitle FranchiseTitle {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_FranchiseTitle_get(swigCPtr);
      GnTitle ret = (cPtr == IntPtr.Zero) ? null : new GnTitle(cPtr, true);
      return ret;
    } 
  }

  public GnExternalIdEnumerable ExternalIds {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_ExternalIds_get(swigCPtr);
      GnExternalIdEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnExternalIdEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoWorkEnumerable Works {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Works_get(swigCPtr);
      GnVideoWorkEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoWorkEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoProductEnumerable Products {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Products_get(swigCPtr);
      GnVideoProductEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoProductEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoCreditEnumerable VideoCredits {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_VideoCredits_get(swigCPtr);
      GnVideoCreditEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoCreditEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoSeriesEnumerable Series {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Series_get(swigCPtr);
      GnVideoSeriesEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoSeriesEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnContentEnumerable Contents {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoSeason_Contents_get(swigCPtr);
      GnContentEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnContentEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnVideoSeason.java

package com.gracenote.gnsdk;

/** 
**/ 
 
public class GnVideoSeason extends GnDataObject {
  private long swigCPtr;

  protected GnVideoSeason(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnVideoSeason_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnVideoSeason obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnVideoSeason(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnVideoSeason_gnType();
  }

  public static GnVideoSeason from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnVideoSeason(gnsdk_javaJNI.GnVideoSeason_from(GnDataObject.getCPtr(obj), obj), true);
  }

  public GnVideoSeason(String id, String idTag) throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnVideoSeason(id, idTag), true);
  }

/** 
*  Flag indicating if response result contains full (true) or partial metadata. 
* <p><b>Note:</b></p> 
*   What constitutes a full result varies among response results, and also 
*  depends on data availability. 
* 
*/ 
 
  public boolean isFullResult() {
    return gnsdk_javaJNI.GnVideoSeason_isFullResult(swigCPtr, this);
  }

/** 
*  Video season's Gracenote identifier. 
* 
*/ 
 
  public String gnId() {
    return gnsdk_javaJNI.GnVideoSeason_gnId(swigCPtr, this);
  }

/** 
*   Video season's Gracenote unique identifier. 
* 
*/ 
 
  public String gnUId() {
    return gnsdk_javaJNI.GnVideoSeason_gnUId(swigCPtr, this);
  }

/** 
*  Video season's product ID (aka Tag ID). 
*  <p><b>Remarks:</b></p> 
*  Available for most types, this value can be stored or transmitted. Can 
*  be used as a static identifier for the current content as it will not change over time. 
* 
*/ 
 
  public String productId() {
    return gnsdk_javaJNI.GnVideoSeason_productId(swigCPtr, this);
  }

/** 
*  Video season's TUI (title-unique identifier). 
* 
*/ 
 
  public String tui() {
    return gnsdk_javaJNI.GnVideoSeason_tui(swigCPtr, this);
  }

/** 
*  Video season's Tui Tag. 
* 
*/ 
 
  public String tuiTag() {
    return gnsdk_javaJNI.GnVideoSeason_tuiTag(swigCPtr, this);
  }

/** 
*  Video production type, e.g., Animation. This is a list/locale dependent value. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful 
*  allocation of a {@link GnLocale} object 
* <p> 
*  To render locale-dependent information for list-based values, the application must allocate a {@link GnLocale} object. 
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for 
*  a list-based value. 
* 
*/ 
 
  public String videoProductionType() {
    return gnsdk_javaJNI.GnVideoSeason_videoProductionType(swigCPtr, this);
  }

/** 
*  Video production type identifier. 
* 
*/ 
 
  public long videoProductionTypeId() {
    return gnsdk_javaJNI.GnVideoSeason_videoProductionTypeId(swigCPtr, this);
  }

/** 
*  Video season's original release date. 
* 
*/ 
 
  public String dateOriginalRelease() {
    return gnsdk_javaJNI.GnVideoSeason_dateOriginalRelease(swigCPtr, this);
  }

/** 
*  Duration in seconds such as "3600" for a 60-minute program. 
* 
*/ 
 
  public long duration() {
    return gnsdk_javaJNI.GnVideoSeason_duration(swigCPtr, this);
  }

/** 
*  Duration units value (seconds, "SEC"). 
* 
*/ 
 
  public String durationUnits() {
    return gnsdk_javaJNI.GnVideoSeason_durationUnits(swigCPtr, this);
  }

/** 
*  Franchise number. 
* 
*/ 
 
  public long franchiseNum() {
    return gnsdk_javaJNI.GnVideoSeason_franchiseNum(swigCPtr, this);
  }

/** 
*  Franchise count. 
* 
*/ 
 
  public long franchiseCount() {
    return gnsdk_javaJNI.GnVideoSeason_franchiseCount(swigCPtr, this);
  }

/** 
*  Plot sypnosis, e.g., for Friends episode:."Monica's popularity at a karaoke club might have more to do with her revealing dress than her voice" 
* 
*/ 
 
  public String plotSynopsis() {
    return gnsdk_javaJNI.GnVideoSeason_plotSynopsis(swigCPtr, this);
  }

/** 
*  Plot sypnosis language, e.g., English 
* 
*/ 
 
  public String plotSynopsisLanguage() {
    return gnsdk_javaJNI.GnVideoSeason_plotSynopsisLanguage(swigCPtr, this);
  }

/** 
*  Plot tagline, e.g., "An adventure as big as life itself." 
* 
*/ 
 
  public String plotTagline() {
    return gnsdk_javaJNI.GnVideoSeason_plotTagline(swigCPtr, this);
  }

/** 
*  Serial type, e.g., Series or Episode 
* 
*/ 
 
  public String serialType() {
    return gnsdk_javaJNI.GnVideoSeason_serialType(swigCPtr, this);
  }

/** 
*  Work type, e.g., Musical 
* 
*/ 
 
  public String workType() {
    return gnsdk_javaJNI.GnVideoSeason_workType(swigCPtr, this);
  }

/** 
*  Target audience, e.g., "Kids and Family" 
* 
*/ 
 
  public String audience() {
    return gnsdk_javaJNI.GnVideoSeason_audience(swigCPtr, this);
  }

/** 
*  Video mood, e.g., Playful. 
* 
*/ 
 
  public String videoMood() {
    return gnsdk_javaJNI.GnVideoSeason_videoMood(swigCPtr, this);
  }

/** 
*  Overall story type, e.g., "Love Story". 
* 
*/ 
 
  public String storyType() {
    return gnsdk_javaJNI.GnVideoSeason_storyType(swigCPtr, this);
  }

/** 
*  Reputation, e.g., "Chick flick". 
* 
*/ 
 
  public String reputation() {
    return gnsdk_javaJNI.GnVideoSeason_reputation(swigCPtr, this);
  }

/** 
*  Scenario, e.g., Action 
* 
*/ 
 
  public String scenario() {
    return gnsdk_javaJNI.GnVideoSeason_scenario(swigCPtr, this);
  }

/** 
*  Setting environment, e.g., "High School" 
* 
*/ 
 
  public String settingEnvironment() {
    return gnsdk_javaJNI.GnVideoSeason_settingEnvironment(swigCPtr, this);
  }

/** 
*  Setting time period, e.g., "Jazz Age 1919-1929". 
* 
*/ 
 
  public String settingTimePeriod() {
    return gnsdk_javaJNI.GnVideoSeason_settingTimePeriod(swigCPtr, this);
  }

/** 
*  Topic value, e.g., "Teen Angst" 
* 
*/ 
 
  public String topic() {
    return gnsdk_javaJNI.GnVideoSeason_topic(swigCPtr, this);
  }

/** 
*  Season number. 
* 
*/ 
 
  public long seasonNumber() {
    return gnsdk_javaJNI.GnVideoSeason_seasonNumber(swigCPtr, this);
  }

/** 
*  Season count - total number of seasons. 
* 
*/ 
 
  public long seasonCount() {
    return gnsdk_javaJNI.GnVideoSeason_seasonCount(swigCPtr, this);
  }

/** 
*  Source, e.g., "Fairy Tales and Nursery Rhymes" 
* 
*/ 
 
  public String source() {
    return gnsdk_javaJNI.GnVideoSeason_source(swigCPtr, this);
  }

/** 
*  Style, e.g., "Film Noir" 
* 
*/ 
 
  public String style() {
    return gnsdk_javaJNI.GnVideoSeason_style(swigCPtr, this);
  }

/** 
*  Genre. This is a list/locale dependent, multi-level field. 
* <p> 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful 
*  allocation of a <code>GnLocale</code> object 
* <p> 
*  To render locale-dependent information for list-based values, the application must allocate a {@link GnLocale} object. 
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for 
*  a list-based value. 
* 
*/ 
 
  public String genre(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoSeason_genre(swigCPtr, this, level.swigValue());
  }

/** 
*  Origin, e.g., "New York City." List/locale dependent, multi-level field. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that the list or locale be loaded into memory through a successful 
*  allocation of a <code>GnLocale</code> object 
* <p> 
*  To render locale-dependent information for list-based values, the application must allocate a {@link GnLocale} object. 
*  The SDK returns a <code>LocaleNotLoaded</code> message when locale information is not set prior to a request for 
*  a list-based value. 
* 
*/ 
 
  public String origin(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoSeason_origin(swigCPtr, this, level.swigValue());
  }

/** 
*  Rating object. 
* 
*/ 
 
  public GnRating rating() {
    return new GnRating(gnsdk_javaJNI.GnVideoSeason_rating(swigCPtr, this), true);
  }

/** 
*   Official title object 
* 
*/ 
 
  public GnTitle officialTitle() {
    return new GnTitle(gnsdk_javaJNI.GnVideoSeason_officialTitle(swigCPtr, this), true);
  }

/** 
*  Franchise title object. 
* 
*/ 
 
  public GnTitle franchiseTitle() {
    return new GnTitle(gnsdk_javaJNI.GnVideoSeason_franchiseTitle(swigCPtr, this), true);
  }

  public GnExternalIdIterable externalIds() {
    return new GnExternalIdIterable(gnsdk_javaJNI.GnVideoSeason_externalIds(swigCPtr, this), true);
  }

  public GnVideoWorkIterable works() {
    return new GnVideoWorkIterable(gnsdk_javaJNI.GnVideoSeason_works(swigCPtr, this), true);
  }

  public GnVideoProductIterable products() {
    return new GnVideoProductIterable(gnsdk_javaJNI.GnVideoSeason_products(swigCPtr, this), true);
  }

  public GnVideoCreditIterable videoCredits() {
    return new GnVideoCreditIterable(gnsdk_javaJNI.GnVideoSeason_videoCredits(swigCPtr, this), true);
  }

  public GnContentIterable contents() {
    return new GnContentIterable(gnsdk_javaJNI.GnVideoSeason_contents(swigCPtr, this), true);
  }

  public GnVideoSeriesIterable series() {
    return new GnVideoSeriesIterable(gnsdk_javaJNI.GnVideoSeason_series(swigCPtr, this), true);
  }

}
